setup/src/Magento/Setup/Model/ModuleUninstaller.php

```diff
@@ -94,6 +95,19 @@ public function uninstallData(OutputInterface $output, array $modules)
             $this->getPatchApplier()->revertDataPatches($module);
         }
     }
+    
+    /**
+     * Checks wether auth.json exists in the Magento root.
+     * @throws FileNotFoundException when auth.json is not found with a reference to the developer documentation.
+     */
+    private function failOnMissingAuthJson() 
+    {
+        $auth_json_path = BP . DIRECTORY_SEPARATOR . 'auth.json';
+        if (!file_exists($auth_json_path)) {
+            $error_message = sprintf('auth.json at %s could not be found. Refer to https://devdocs.magento.com/guides/v2.4/install-gde/prereq/dev_install.html#authentication-file how to install this file.', $auth_json_path);
+            throw new FileNotFoundException($error_message);
+        }
+    }
```

Comment:
Are you sure it's ok to make the auth.json mandatory? I don't think so, as we might set the credentials globally.

Reply:
I have set my auth.json globally, via ~/.composer/auth.json and that works fine when a manual `~ composer update` or `~ composer remove xxx` is run. But when this composer sub process is started up, via the `~ bin/magento module:uninstall` it doesn't read from ~/.composer/auth.json for whatever reason, which then leads to the bug as is set out in the related issue as it prompts for the password behind the scenes. It only continues on when the data is present in the auth.json file.

I've made a dummy repo here of the plugin that demonstrates the issue. https://github.com/tschallacka/mage2-dummy-test

When there is an auth.json in the root it runs fine, as is highlighted in my blog post. If there is no auth.json in the root, it hangs and gives issues.

Reply:
It looks like the issue is related to incorrect composer usage, so it's not reading the global auth.json file. The correct fix should fix this inconsistent behavior.

Reply:
I'd be amenable to make a more comprehensive test if you could point me to where the wrapped composer instance tries to read auth.json and when it fails, if you have a better suggested course of action, to prevent the hanging screen i'm open. Just an error message like in my plugin perhaps?

Reply:
I don't know exactly where that code located, but it looks like the composer remove logic is here: https://github.com/magento/magento2/blob/8b9fff56f2172bb2e3fa5bde4d16daa18df97e22/lib/internal/Magento/Framework/Composer/Remove.php#L41-L52
Please set the breakpoint on this code and debug deeper.

Reply:
> It looks like the issue is related to incorrect composer usage, so it's not reading the global auth.json file. The correct fix should fix this inconsistent behavior.

totally agree, that composer should read auth.json anywhere

Reply:
@ihor-sviziev if you read my blog post at https://blog.tschallacka.de/2021/05/binmagento-moduleuninstall-never.html you can follow my tracking process how I traced the composer path related to the output. I never delved into the actual process of composer itself as it takes so many twists and turns that after 15 minutes of pressing step into I still wasn't at the auth.json search steps. I must admit I kinda gave up.